From 87376dc80a586e51fe637cab61aeaa281960cc86 Mon Sep 17 00:00:00 2001
From: Nikita Shubin <nikita.shubin@kemeke.ru>
Date: Tue, 19 Jul 2016 13:26:51 +0300
Subject: [PATCH] Added ivshmem pci subids to parameter list

---
 hw/misc/ivshmem.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/hw/misc/ivshmem.c b/hw/misc/ivshmem.c
index dcfc8cc..565c15b 100644
--- a/hw/misc/ivshmem.c
+++ b/hw/misc/ivshmem.c
@@ -101,6 +101,10 @@ typedef struct IVShmemState {
     uint32_t features;
     MSIVector *msi_vectors;
 
+    /* subsystem and subvendor ids */
+    uint16_t        subsystem_vendor_id;
+    uint16_t        subsystem_id;
+
     Error *migration_blocker;
 
     char * shmobj;
@@ -910,6 +914,9 @@ static void pci_ivshmem_realize(PCIDevice *dev, Error **errp)
     pci_conf = dev->config;
     pci_conf[PCI_COMMAND] = PCI_COMMAND_IO | PCI_COMMAND_MEMORY;
 
+    pci_set_word(pci_conf + PCI_SUBSYSTEM_VENDOR_ID, s->subsystem_vendor_id);
+    pci_set_word(pci_conf + PCI_SUBSYSTEM_ID, s->subsystem_id);
+
     pci_config_set_interrupt_pin(pci_conf, 1);
 
     memory_region_init_io(&s->ivshmem_mmio, OBJECT(s), &ivshmem_mmio_ops, s,
@@ -1137,6 +1144,8 @@ static Property ivshmem_properties[] = {
     DEFINE_PROP_STRING("shm", IVShmemState, shmobj),
     DEFINE_PROP_STRING("role", IVShmemState, role),
     DEFINE_PROP_UINT32("use64", IVShmemState, ivshmem_64bit, 1),
+    DEFINE_PROP_UINT16("sub-vendor-id", IVShmemState, subsystem_vendor_id, PCI_VENDOR_ID_IVSHMEM),
+    DEFINE_PROP_UINT16("sub-device-id", IVShmemState, subsystem_id, PCI_DEVICE_ID_IVSHMEM),
     DEFINE_PROP_END_OF_LIST(),
 };
 
-- 
2.3.6

